src/energyplus/ForwardTranslator/ForwardTranslateElectricEquipment.cpp

```diff
@@ -57,23 +57,25 @@ namespace openstudio {
 namespace energyplus {
 
   boost::optional<IdfObject> ForwardTranslator::translateElectricEquipment(ElectricEquipment& modelObject) {
-    IdfObject idfObject(openstudio::IddObjectType::ElectricEquipment);
-    m_idfObjects.push_back(idfObject);
+
+    IdfObject idfObject = createRegisterAndNameIdfObject(openstudio::IddObjectType::ElectricEquipment, modelObject);
```

Comment:
yes, I think createRegisterAndNameIdfObject is the correct way. I'm probably the most guilty party of not using this method, although in this case I don't think it was me. I think it is more common in the oldest type translations, which were written before the createRegisterAndName method existed.

Reply:
The GenerateClass.rb portion that adds automatic FT code is quite new too, I added that a year or two ago, so there were no boilerplate code in the past.
The automatic FT does createNameAndRegister by default but has a TODO with commented code to explain that you could do it differently in the case where you might actually test for a condition where the object shouldn't be translated (missing required children or invalid combinations)